pkg/app/piped/cloudprovider/kubernetes/state.go

```diff
@@ -534,3 +536,44 @@ func determineServiceAccountHealth(obj *unstructured.Unstructured) (status model
 	status = model.KubernetesResourceState_HEALTHY
 	return
 }
+
+func determineNameSpace(obj *unstructured.Unstructured) (status model.KubernetesResourceState_HealthStatus, desc string) {
+	ns := &corev1.Namespace{}
+	err := scheme.Scheme.Convert(obj, ns, nil)
+	if err != nil {
+		status = model.KubernetesResourceState_OTHER
+		desc = fmt.Sprintf("Unexpected error while calculating: unable to convert %T to %T: %v", obj, ns, err)
+		return
+	}
+
+	switch ns.Status.Phase {
+	case corev1.NamespaceActive:
+		// Correct
+	case corev1.NamespaceTerminating:
+		status = model.KubernetesResourceState_OTHER
+		desc = "NameSpace is gracefully terminated"
+	default:
+		status = model.KubernetesResourceState_OTHER
+		desc = "The current phase of NameSpace is unexpected"
+	}
+
+	status = model.KubernetesResourceState_HEALTHY
```

Comment:
Won't this override the value configured at line 553 or line 556?

Reply:
Yes, it is. Let me modify it.